### PR TITLE
Project list incorrect status

### PR DESCRIPTION
**Pull request by andrestack**

### CODE REVIEW

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-7)

File: cypress/e2e/project-list.cy.ts [hunk 1]
```diff
@@ -20,19 +19,60 @@ describe("Project List", () => {
...
-          cy.wrap($el).contains(mockProjects[index].name);
+          cy.wrap($el).contains(projects.name);
           cy.wrap($el).contains(languageNames[index]);
-          cy.wrap($el).contains(mockProjects[index].numIssues);
-          cy.wrap($el).contains(mockProjects[index].numEvents24h);
-          cy.wrap($el).contains(capitalize(mockProjects[index].status));
+          cy.wrap($el).contains(projects.numIssues);
+          cy.wrap($el).contains(projects.numEvents24h);
+          cy.wrap($el).contains(expectedStatus[status]);
+          // cy.wrap($el).contains(capitalize(mockProjects[index].status));
```
Comment (romanchukenator) [inline cypress/e2e/project-list.cy.ts]: If it's not needed remove it?

**Review by romanchukenator — APPROVED** [review]
Awesome. LGTM. Ship it!

File: cypress/e2e/project-list.cy.ts [hunk 2]
```diff
@@ -20,19 +19,60 @@ describe("Project List", () => {
...
+        info: "rgb(236, 253, 243)",
+      };
+
       // get all project cards
       cy.get("main")
         .find("li")
         .each(($el, index) => {
+          type StatusKey = "error" | "warning" | "info";
+          const status: StatusKey = mockProjects[index].status as StatusKey;
+          const projects = mockProjects[index];
```
Comment (jkettmann) [inline cypress/e2e/project-list.cy.ts]: Nit: I'd rename this to project as it's only one

File: cypress/e2e/project-list.cy.ts [hunk 3]
```diff
@@ -20,19 +19,60 @@ describe("Project List", () => {
...
+        warning: "rgb(255, 250, 235)",
+        info: "rgb(236, 253, 243)",
+      };
+
       // get all project cards
       cy.get("main")
         .find("li")
         .each(($el, index) => {
+          type StatusKey = "error" | "warning" | "info";
+          const status: StatusKey = mockProjects[index].status as StatusKey;
```
Comment (jkettmann) [inline cypress/e2e/project-list.cy.ts]: I think `const status = ...` is sufficient as you convert the string already with the ` as StatusKey`

Comment (jkettmann) [reply cypress/e2e/project-list.cy.ts]: Actually can't you directly convert with `as Status` or maybe `as keyof Status` to reuse the Status enum?

File: features/ui/badge/badge.tsx [hunk 4]
```diff
@@ -26,9 +26,12 @@ export function Badge({
   children,
   size = BadgeSize.md,
   color = BadgeColor.primary,
-}: BadgeProps) {
+}: BadgeProps): React.JSX.Element {
```
Comment (jkettmann) [inline features/ui/badge/badge.tsx:29]: I don't think this is required?! The type should be inferred from the return value. As that's a JSX element all should be fine without typing the return value

**Review by jkettmann — APPROVED** [review]
Just a few minor things. Really good work. Clean and solid

File: cypress/e2e/project-list.cy.ts (hunk 3 above)
Comment (andrestack) [reply cypress/e2e/project-list.cy.ts]: Absolutely! If the enum is already declared, no need to repeat it. I imported it to the the project-list.cy.ts file and defined the `type StatusKey` to include all the keys of `ProjectStatus`

**Pull request merged**